Nuxt 3 Dependency Updates

treckstar opened the pull request:
Uniform + Nuxt 3 is a great combo. Since I made these updates while testing I figured I would send them over.

Thanks!

treckstar committed:
fix: default env example variables

treckstar committed:
fix: node 16 past end of life

treckstar committed:
fix: pnpm & all dependencies latest version

Additionally, cleared warning about missing dependencies
that are supposed to be optimized. These have been added to
the `devDependencies`.

treckstar committed:
fix: nuxt 3 convention for catch-all page

treckstar committed:
fix: ignoring files based on nuxt 3 docs

treckstar committed:
fix: section for visual editing with nuxt, preview url

treckstar committed:
fix: postcss now lives in `nuxt.config.ts`

This change silences the warning about PostCSS on dev/build.
Nuxt uses `nuxt.config.ts` as single source of truth for
configuration and in some cases skips external config files.

treckstar committed:
fix: .env to preview

treckstar committed:
chore(dependencies): update uniform to 19.92.2

treckstar committed:
chore(dependencies): update uniform to 19.94.0

treckstar committed:
chore(dependencies): update uniform to 19.96.0

treckstar committed:
chore(dependencies): lockfile

treckstar committed:
chore(deps): uniform 19.100.0

renovate[bot] committed:
fix(deps): update uniform to v19.106.1

treckstar committed:
chore(deps): uniform 19.106.1

treckstar committed:
chore(deps): nuxt 3.10.0

alexshyba commented:
Thank you so much for the first external contribution to our starters, @treckstar! Really appreciate it. Tagging my colleague @ilyasmez to help review this PR.

ilyasmez commented on README.md:
```diff
@@ -9,7 +9,15 @@ https://www.npmjs.com/package/@uniformdev/uniform-nuxt
+
+### Setup Uniform Preview URL for Visual Editing
+
+Make sure to [set the preview URL in the Uniform Canvas Settings](https://docs.uniform.app/docs/guides/composition/visual-editing/preview-mode#configure-the-preview-url-for-the-project) to the value for Nuxt.
```
❤️

ilyasmez commented on nuxt.config.ts:
```diff
@@ -8,9 +8,9 @@ export default defineNuxtConfig({
-    apiHost: process.env.UNIFORM_CLI_BASE_URL,
-    edgeApiHost: process.env.UNIFORM_CLI_BASE_EDGE_URL,
-    outputType: process.env.OUTPUT_TYPE,
+    apiHost: process.env.UNIFORM_CLI_BASE_URL || "https://uniform.app",
+    edgeApiHost: process.env.UNIFORM_CLI_BASE_EDGE_URL || "https://uniform.global",
+    outputType: process.env.OUTPUT_TYPE || 'standard',
```
These are already set as default values on the module-level. Did you face any issues with that?

Or you want to handle empty strings? If that's the case, you can just set the falsy values to undefined (using `|| undefined,`) to leverage the module's defaults.

treckstar replied:
Done!

ilyasmez commented on .nvmrc:
```diff
@@ -1 +1 @@
-v16.15.0
\ No newline at end of file
+v18.18.0
```
❤️

ilyasmez reviewed (commented):
Thank you very much for taking the time to open a PR, @treckstar!! ⭐

The PR looks great! We just need to fix 3 things before merging:
- Bring back the `.env` file: We need to bring back the .env file because it's used for an automated deployment of this project (yes, we could create one out of the .env.example, but that's not implemented yet.)
- gitignore the lock file: This goes against the best practices for sure, but it's required for the internal deployment of the repo (in addition to some issues related to running the project in StackBlitz).
- Use `npm` instead of `pnpm`: We're big fans of pnpm and we use it a lot internally, but to keep the project more accessible, we prefer to use the standard package manager.

treckstar committed:
fix: used npm instead of pnpm

treckstar committed:
fix: deleted lock file from repo

Important for internal deployment purposes.

treckstar committed:
fix: ignored npm lock file & allow .env

treckstar committed:
fix: removed for do not repeat yourself

These default values already are in the codebase, and 
setting to || undefined will set them as default when not 
specified.

treckstar commented:
Thank you for taking the time to review! Much appreciated :)

I think I touched on all of those changes. I did pull down and it builds locally. I think Netlify needs to clear cache? I also got that error, and once `rimraf node_modules` was done, it worked just fine. 

If you'd like me to clean this branch up and rebase to 1 commit or anything at all. Just let me know!

treckstar committed:
fix: included option for npm legacy deps

ilyasmez reviewed (approved):
Perfect! Thank you very much for your contribution @treckstar! 🙏 

And yes, clearing the cache on Netlify did the job, your PR was successfully deployed here: https://deploy-preview-62--uniformconf-nuxt.netlify.app/
